lib/_sendEvent.ts

```diff
@@ -57,6 +57,25 @@ export function makeSendEvents(requestFn: RequestFnType) {
       this._apiKey,
       additionalParams?.headers
     );
+    return isPromise(send) ? send.then(purgePurchased(events)) : send;
```

Comment:
Just so I'm clear, what's the reasoning behind not purging in `purchasedObjectIDs(AfterSearch)` seeing as, if I'm understanding correctly, that's the level of abstraction we're doing the storing?

Reply:
I made the choice to put it here because once an object is purchased (no matter the method used to mark the object as purchased) I want it removed from the object/query cache. Our customers may use either `purchasedObjectIDs` or `purchasedObjectIDsAfterSearch` (they could even use `convertedObjectIDs` or `convertedObjectIDsAfterSearch` if they're not using TS and it would still work if they're passing in the `eventSubtype` param) and I wanted to ensure it covered all of these cases.